rails_event_store_active_record/lib/rails_event_store_active_record/event_repository.rb

```diff
@@ -1,14 +1,51 @@
+require 'activerecord-import'
+
 module RailsEventStoreActiveRecord
   class EventRepository
     def initialize(adapter: Event)
       @adapter = adapter
     end
     attr_reader :adapter
 
-    def create(event, stream_name)
-      data = event.to_h.merge!(stream: stream_name, event_type: event.class)
-      adapter.create(data)
-      event
+    def append_to_stream(events, stream_name, expected_version)
+      events = [*events]
+      expected_version   = case expected_version
```

Comment:
this should be in its own private method. It makes the `append_to_stream` very hard to reason about otherwise, also violates single responsibility principle